src/components/RadioInputBox/RadioInputBox.tsx

```diff
@@ -1,51 +1,155 @@
-import { useState } from "react";
+import { useEffect, useRef, useState } from "react";
+import classnames from "classnames";
 
-import type { ActionOptions } from "panels/ActionsPanel/ActionsPanel";
+import type {
+  ActionOptions,
+  SetSelectedAction,
+} from "panels/ActionsPanel/ActionsPanel";
+
+import DescriptionSummary from "./DescriptionSummary";
 
 import "./_radio-input-box.scss";
 
 type Props = {
   name: string;
   description: string;
   options: ActionOptions;
+  selectedAction: string | undefined;
+  onSelect: SetSelectedAction;
 };
 
 export default function RadioInputBox({
   name,
   description,
   options,
+  selectedAction,
+  onSelect,
 }: Props): JSX.Element {
   const [opened, setOpened] = useState<boolean>(false);
+  const inputBoxRef = useRef<HTMLDivElement>(null);
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  // 20, 40 are magic numbers that align nicely with the heights.
+  const initialHeight = 40;
+  const paddingNumber = 20;
+
+  useEffect(() => {
+    if (inputBoxRef.current !== null) {
+      // Due to the 'closed' hook animation running on initial load we cannot
+      // set the initial height in the css but have to do it on the first load.
+      inputBoxRef.current.style.height = initialHeight + "px";
+    }
+  }, [inputBoxRef]);
+
+  useEffect(() => {
+    setOpened(selectedAction === name);
+  }, [selectedAction, name]);
+
+  useEffect(() => {
+    const wrapper = inputBoxRef.current;
+    const container = containerRef.current;
 
-  const handleSelect = (e: any) => {
-    const bool = e.target.value === "on" ? true : false;
-    setOpened(bool);
+    if (wrapper === null || container === null) return;
+
+    let startHeight = 0;
+    let endHeight = 0;
+    let duration = 0;
+
+    if (opened) {
+      startHeight = wrapper.offsetHeight;
+      // To be used when we're closing so we know what the original value was.
+      endHeight = container.offsetHeight + paddingNumber;
+      duration = endHeight;
+      // Set the height of the wrapper element to the end height to
+      // override the height set in css for the collapsed size.
+      wrapper.style.height = endHeight + "px";
+    } else if (wrapper.offsetHeight !== initialHeight) {
+      // Do not animate if the wrapper height is already in the closed state
+      // and this hook runs for the closed state again.
+      endHeight = initialHeight;
+      startHeight = container.offsetHeight + paddingNumber;
+      duration = startHeight;
+    }
+    const animation = wrapper.animate(
```

Comment:
This is using [WAAPI](https://developer.mozilla.org/en-US/docs/Web/API/Web_Animations_API/Web_Animations_API_Concepts) so it's still as performant as doing it in CSS but this actually works 😆